### PR TITLE
Conflict with symfony/symfony (any version) to prevent project self-destruct

### PR DESCRIPTION
**Pull request by teohhanhui**
Lesson learnt the hard way.

Installing a bundle that required `symfony/symfony` resulted in catastrophic auto-unconfiguring. Even `etc/packages/app.yaml` and `etc/container.yaml` are gone (as they are from the `symfony/framework-bundle` recipe).

### CODE REVIEW

**Comment by teohhanhui** [comment]
ping @fabpot @Pierstoval

**Comment by Pierstoval** [comment]
TL;DR:
We should add a feature in Flex that will retrieve recipes for any package in the `replace` composer option for any installed package.

---

I'm not sure I would approve this. By opinion, I approve, but for related technical stuff, I think we should allow users to use `symfony/symfony`, the only problem is to handle the manifest for this project, or update Flex so it can manage packages that `replace` other packages, which is the case for `symfony/symfony`, but also for other packages that may need this feature too.

Actually, any monolithic project splitted in subtrees should benefit from a feature coming from Flex, not be restricted to like this PR proposes.

**Comment by fabpot** [comment]
Thank you @teohhanhui.

**Comment by lyrixx** [comment]
Do that mean we can not require symfony/symfony anymore? 
I think it's not very cool. I don't want to add each component one by one in a new project.
May be I'm a bit old fashion, But I prefer to get all components at once.

**Comment by teohhanhui** [comment]
@Pierstoval @lyrixx I completely agree. But this is the short-term solution to prevent catastrophic failures from besetting unsuspecting users.
